simln-lib/src/sim_node.rs

```diff
@@ -1477,7 +1477,7 @@ async fn propagate_payment(request: PropagatePaymentRequest) {
             );
             PaymentResult {
                 htlc_count: 0,
-                payment_outcome: PaymentOutcome::Unknown,
+                payment_outcome: PaymentOutcome::IndexFailure(fail_idx.unwrap_or(0)),
```

Comment:
Let's sanity check the values we're using here. 

AFAIK for a payment sent from `A -> B -> C` the values mean the following:
- `0`: Failed at A
- `1`: Failed at B
- `2`: Failed at C

So when we have `None`, it makes sense that we `unwrap` to zero. What I want to check is when we return `Some(0)` in the simulation - did that fail on `A` or `B`?